climada/hazard/centroids/centr.py

```diff
@@ -79,280 +52,215 @@ class Centroids():
-            Defaults to empty dict()
-        area_pixel : np.array, optional
-            area of size size. Defaults to empty array
-        on_land : np.array, optional
-            on land (True) and on sea (False) of size size. Defaults to empty array
+        crs : str
+            coordinate reference system
         region_id : np.array, optional
-            country region code of size size, Defaults to empty array
-        elevation : np.array, optional
-            elevation of size size. Defaults to empty array
-        dist_coast : np.array, optional
-            distance to coast of size size. Defaults to empty array
+            country region code of size size, Defaults to None array
+        on_land : np.array, optional
+            on land (True) and on sea (False) of size size. Defaults to None array
         """
 
-        self.lat = lat if lat is not None else np.array([])
-        self.lon = lon if lon is not None else np.array([])
-        self.geometry = geometry if geometry is not None else gpd.GeoSeries(crs=DEF_CRS)
-        self.meta = meta if meta is not None else dict()
-        self.area_pixel = area_pixel if area_pixel is not None else np.array([])
-        self.on_land = on_land if on_land is not None else np.array([])
-        self.region_id = region_id if region_id is not None else np.array([])
-        self.elevation = elevation if elevation is not None else np.array([])
-        self.dist_coast = dist_coast if dist_coast is not None else np.array([])
-
-    def check(self):
-        """Check integrity of stored information.
-
-        Checks that either `meta` attribute is set, or `lat`, `lon` and `geometry.crs`.
-        Checks sizes of (optional) data attributes."""
-        n_centr = self.size
-        for var_name, var_val in self.__dict__.items():
-            if var_name in self.vars_check:
-                if var_val.size > 0 and var_val.size != n_centr:
-                    raise ValueError(f'Wrong {var_name} size: {n_centr} != {var_val.size}.')
-        if self.meta:
-            for name in ['width', 'height', 'crs', 'transform']:
-                if name not in self.meta.keys():
-                    raise ValueError('Missing meta information: %s' % name)
-            xres, xshear, _xoff, yshear, yres, _yoff = self.meta['transform'][:6]
-            if xshear != 0 or yshear != 0:
-                raise ValueError('Affine transformations with shearing components are not '
-                                 'supported.')
-            if yres > 0 or xres < 0:
-                raise ValueError('Affine transformations with positive y-orientation '
-                                 'or negative x-orientation are not supported.')
-
-    def equal(self, centr):
+        attr_dict = {
+            'geometry': gpd.points_from_xy(lon, lat, crs=crs),
+            'region_id': region_id,
+            'on_land': on_land
+        }
+        self.gdf = gpd.GeoDataFrame(data=attr_dict)
```

Comment:
Maybe make this attribute private with `self._gdf`? This way we show users that we don't want them to modify it themselves.

Reply:
I am not sure this is necessary.

Reply:
If you want the gdf to be never modified by a user, it should be marked as private. This will also help with tracking bugs, as the linter will raise issues when a private attribute is accessed from outside the class.

Reply:
The question is: are we really sure that we do not want a user at all to be able to add columns to the geodataframe for convenience?

Reply:
SO, I have tried it, but this creates several legacy issues (that can be fixed with enough effort). I am not yet convinced that making the gdf attribute fully private is the right way to go. Thus, before doing this, we need to be very sure that this is the best imho. At the moment I don't see it. The disadvantages (legacy, removal of some flexibility, extra work) seem not to be worth it for me.

Reply:
I also think we are mixing two things here. We should first aim for a consistent use of the GeoDataFrame while maintaining the original interface and functionality as far as possible. Once this is done, we can talk about a new, more handy interface of the Centroids class.

Reply:
That makes sense. But so, should we first make it private, or first leave it as is and consider this later? 

Note however that we won't have much capacity after this work to further improve the Centroid class since there are many other important things to do, so we should still have a very functional version.